example-app/src/main/java/com/adyen/checkout/example/service/ExampleDropInService.kt

```diff
@@ -40,45 +41,49 @@ class ExampleDropInService : DropInService() {
     private val paymentsRepository: PaymentsRepository by inject()
     private val keyValueStorage: KeyValueStorage by inject()
 
-    override fun makePaymentsCall(paymentComponentData: JSONObject): DropInServiceResult {
-        Logger.d(TAG, "makePaymentsCall")
-
-        // Check out the documentation of this method on the parent DropInService class
-        val paymentRequest = createPaymentRequest(
-            paymentComponentData,
-            keyValueStorage.getShopperReference(),
-            keyValueStorage.getAmount(),
-            keyValueStorage.getCountry(),
-            keyValueStorage.getMerchantAccount(),
-            RedirectComponent.getReturnUrl(applicationContext),
-            AdditionalData(
-                allow3DS2 = keyValueStorage.isThreeds2Enable().toString(),
-                executeThreeD = keyValueStorage.isExecuteThreeD().toString()
+    override fun makePaymentsCall(paymentComponentData: JSONObject) {
+        ThreadManager.EXECUTOR.submit {
```

Comment:
@caiofaustino as opposed to an `IntentService`, this `Service` will run on the main thread by default, and it's up to the merchants to manage that. But I suppose this wouldn't be a problem?

Reply:
Hmm, that's a good point. I think this is something we can also provide as an option inside the class implementation.
Because if we maintain the abstraction from v3 for compatibility, I think that should be on the background thread by default as well to not break compatibility. 

We can discuss a way to provide this with a better approach than `AWAIT`.